src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1150,12 +1150,24 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
 
         private string GetTablePrefix(string databaseName, string schemaName)
         {
-            StringBuilder tablePrefix = new(SqlQueryBuilder.QuoteIdentifier(databaseName));
+            StringBuilder tablePrefix = new();
             if (!string.IsNullOrEmpty(schemaName))
             {
+                // determine schemaName for prefix.
                 schemaName = SqlQueryBuilder.QuoteIdentifier(schemaName);
+            }
+
+            if (!string.IsNullOrEmpty(databaseName))
+            {
+                // determine databaseName for prefix.
+                databaseName = SqlQueryBuilder.QuoteIdentifier(databaseName);
+                tablePrefix.Append(databaseName);
                 tablePrefix.Append($".{schemaName}");
```

Comment:
if schemaName is empty and db is not empty, this will be `[db].`, as opposed to just `[db]` that was there previously...

Reply:
Yes am making this fix and adding appropriate test cases.

Reply:
Should be fixed now.